File: seed/php-sdk/trace/src/V2/Problem/Types/VoidFunctionSignatureThatTakesActualResult.php
<?php

namespace Seed\V2\Problem\Types;

use Seed\Core\Json\JsonSerializableType;
use Seed\Core\Json\JsonProperty;
use Seed\Core\Types\ArrayType;
use Seed\Commons\Types\VariableType;

class VoidFunctionSignatureThatTakesActualResult extends JsonSerializableType
{
    /**
     * @var array<Parameter> $parameters
     */
    #[JsonProperty('parameters'), ArrayType([Parameter::class])]
    public array $parameters;

    /**
     * @var VariableType $actualResultType
     */
    #[JsonProperty('actualResultType')]
    public VariableType $actualResultType;

    /**
     * @param array{
     *   parameters: array<Parameter>,
     *   actualResultType: VariableType,
     * } $values
     */
    public function __construct(
        array $values,
    )
    {
        $this->parameters = $values['parameters'];$this->actualResultType = $values['actualResultType'];
    }

    /**
     * @return string
     */
    public function __toString(): string {
        return $this->toJson();
    }
}

File: seed/ruby-sdk-v2/trace/lib/seed/v_2/v_3/problem/void_function_signature_that_takes_actual_result.rb
# frozen_string_literal: true

module Seed
    module Types
        class VoidFunctionSignatureThatTakesActualResult < Internal::Types::Model
            field :parameters, Internal::Types::Array[Seed::V2::V3::Problem::Parameter], optional: false, nullable: false
            field :actual_result_type, Seed::Commons::VariableType, optional: false, nullable: false

    end
end
